sign/api/api.go

```diff
@@ -0,0 +1,33 @@
+// Package api contains a register of signature algorithms.
+package api
+
+import (
+	"github.com/cloudflare/circl/sign"
+	"github.com/cloudflare/circl/sign/ed25519"
+	"github.com/cloudflare/circl/sign/ed448"
+	"github.com/cloudflare/circl/sign/eddilithium3"
+	"github.com/cloudflare/circl/sign/eddilithium4"
+)
+
+var allSchemes [sign.SchemeCount]sign.Scheme
+var allSchemeNames map[string]sign.Scheme
+
+func init() {
+	allSchemeNames = make(map[string]sign.Scheme)
+	register(ed25519.Scheme)
+	register(ed448.Scheme)
+	register(eddilithium3.Scheme)
+	register(eddilithium4.Scheme)
```

Comment:
Why not `register(sign.Ed25519, ed25519.Scheme)`?  Then there is no need for a public `ID()`, `SchemeID`, etc.

Reply:
@armfazh I removed `SchemeID` without changing the rest of the API at all.